landing_page/spec/models/user_spec.rb

```diff
@@ -0,0 +1,90 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+
+    let (:user){User.new}
+
+    context "don't create a user account with unacceptable credentials" do
+
+        let(:user) do
+            User.new(:first_name => 'Testing', :second_name => 'Tested', :email => 'testing@tested.test',
+                    :password => 'password', :password_confirmation => 'password')
+        end
+
+        it "fails if the password and password confirmation does not match" do
+            user.password_confirmation = "password123"
+            expect(user).to_not be_valid
+        end
+
+        it "fails if the password is less than 6 characters long" do
+            user.password = "pass"
+            expect(user).to_not be_valid
+        end  
+
+        it "fails if no password is present" do
+            user.password = nil
+            expect(user).to_not be_valid
+        end
+
+        it "fails if the email does not match the required format - 1" do
+            user.email = "1.com"
+            expect(user).to_not be_valid
+        end  
+
+        it "fails if the email does not match the required format - 2" do
+            user.email = "@gmail.com"
+            expect(user).to_not be_valid
+        end
+
+        it "fails if the email does not match the required format - 3" do
+            user.email = "1@gmail"
+            expect(user).to_not be_valid
+        end
+
+        it "fails if the email does not match the required format - 4" do
+            user.email = "1"
+            expect(user).to_not be_valid
+        end
```

Comment:
These could be merged into a single block with more than one assertion.

Either this way:
```suggestion
        it "fails if the email is invalid" do
            user.email = "1.com"
            expect(user).to_not be_valid
            
            user.email = "@gmail.com"
            expect(user).to_not be_valid
            
            user.email = "1@gmail"
            expect(user).to_not be_valid
            
            user.email = "1"
            expect(user).to_not be_valid
        end
```

Or this way:
```suggestion
        it "fails if the email is invalid" do
            ["1.com", "@gmail.com", "1@gmail", "1"].each do |email|
                user.email = email
                expect(user).to_not be_valid
            end            
        end
```

Which one do you prefer?

Reply:
I preferred the first suggestion, I just feel like it's easier to read 👍 

Resolved in commit 302892f21532d473cba8b1747c5b20e84a23bf73